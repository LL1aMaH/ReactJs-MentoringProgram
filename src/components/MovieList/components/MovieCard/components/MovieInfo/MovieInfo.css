p {
  margin: 0;
  padding: 0;
}

.card {
  display: flex;
  justify-content: space-between;
  color: var(--primary-font-color);
  padding-top: 10px;
}

.info {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
}

.year {
  border: 1px solid var(--primary-font-color);
  border-radius: 4px;
  padding: 4px;
  height: 28px;
}

.genres {
  overflow: hidden;
  display: -webkit-box;
  -webkit-box-orient: vertical;
  -webkit-line-clamp: 1;
}
