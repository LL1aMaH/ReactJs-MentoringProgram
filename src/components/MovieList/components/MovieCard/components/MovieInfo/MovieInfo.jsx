import React from 'react';

import styles from '../../../../../../../styles/MovieInfo.module.scss'


export const MovieInfo = (props) => {
  const { title, genres, year } = props;
  return (
    <div className={styles.card}>
      <div className={styles.info}>
        <p className={styles.title} id="p1">{title}</p>
        <p className={styles.genres} id="p2">{genres.join(' ')}</p>
      </div>
      <p className={styles.year}>{year}</p>
    </div>
  );
};
