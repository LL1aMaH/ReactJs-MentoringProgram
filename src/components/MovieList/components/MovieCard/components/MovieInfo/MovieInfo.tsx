import React from 'react';

import styles from './MovieInfo.css';

export type MovieCardProps = {
  title: string;
  genres: Array<string>;
  year: number | string;
};

export const MovieInfo = (props: MovieCardProps): JSX.Element => {
  const { title, genres, year } = props;
  return (
    <div className={styles.card}>
      <div className={styles.info}>
        <div className={styles.title}>{title}</div>
        <div className={styles.genres}>{genres.join(' ')}</div>
      </div>
      <div className={styles.year}>{year}</div>
    </div>
  );
};
