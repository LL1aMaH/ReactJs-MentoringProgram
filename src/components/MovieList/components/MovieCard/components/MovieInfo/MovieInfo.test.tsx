import React from 'react';
import { render } from '@testing-library/react';

import { MovieInfo } from './MovieInfo';

describe('MovieInfo', () => {
  it('renders correctly', () => {
    const component = render(<MovieInfo title="title" genres={[]} year="1234" />);
    expect(component.asFragment()).toMatchSnapshot();
  });
});
