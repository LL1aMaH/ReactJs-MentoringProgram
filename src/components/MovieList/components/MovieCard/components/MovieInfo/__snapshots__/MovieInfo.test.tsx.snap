// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieInfo renders correctly 1`] = `
<DocumentFragment>
  <div
    class="card"
  >
    <div
      class="info"
    >
      <p
        class="title"
      >
        title
      </p>
      <p
        class="genres"
      />
    </div>
    <p
      class="year"
    >
      1234
    </p>
  </div>
</DocumentFragment>
`;
